common/tests/interlocked_hl_int/interlocked_hl_int.c

```diff
@@ -99,6 +133,50 @@ TEST_FUNCTION(interlocked_hl_set_and_wake_all_operates_successfully)
     }
 }
 
+/*
+Tests:
+InterlockedHL_WaitForValue64
+InterlockedHL_SetAndWakeAll64
+*/
+static int set_and_wake_all_helper_thread_function_64(void* context)
+{
+    (void)context;
+    ASSERT_ARE_EQUAL(INTERLOCKED_HL_RESULT, INTERLOCKED_HL_OK, InterlockedHL_WaitForValue64(&globalValue64, 15, UINT32_MAX));
+    return 0;
+}
+
+TEST_FUNCTION(interlocked_hl_set_and_wake_all_operates_successfully_64)
+{
+    // + create 10 helper threads which wait on a value
+    // + main thread creates helper threads and sets value for the helper threads to wake up
+    // + main thread joins on the helper threads and ensures that all threads wake up and terminate
+    // arrange
+    globalValue64 = 10;
```

Comment:
>10 [](http://example.com/codeflow?start=20&length=2)

Would be interesting to have this value greater than INT32_MAX, just to see that we exercise the right APIs under the hood.

Reply:
It's covered in INTEG tests